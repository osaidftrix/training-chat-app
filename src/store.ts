import { combineReducers, configureStore } from '@reduxjs/toolkit';
import connectionSlice from './store/slices/connectionSlice';
import storage from 'redux-persist/lib/storage';
//import storage from 'redux-persist-indexeddb-storage';
import { persistReducer, persistStore } from 'redux-persist';
import thunk from 'redux-thunk';
const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['connection'],
  // whitelist: ['connection']
};

const rootReducer = combineReducers({
  connection: connectionSlice,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: [thunk],
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

export const persistor = persistStore(store);
