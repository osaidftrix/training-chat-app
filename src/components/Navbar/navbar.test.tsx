import { describe, test, expect } from 'vitest';
import { Provider } from 'react-redux';
import { persistor, store } from '../../store';
import NavBar from './navbar';
import { MemoryRouter, Routes, Route } from 'react-router';
import { create } from 'react-test-renderer';
import { PersistGate } from 'redux-persist/integration/react';

describe('<NavBar />', () => {
  test('NavBar renders properly', () => {
    const renderer = create(
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <MemoryRouter initialEntries={['/']}>
            <Routes>
              <Route path="/" element={<NavBar />}></Route>
            </Routes>
          </MemoryRouter>
        </PersistGate>
      </Provider>,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
