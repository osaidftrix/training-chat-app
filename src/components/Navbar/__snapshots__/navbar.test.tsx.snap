// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NavBar /> > NavBar renders properly 1`] = `
<nav
  className="bg-white border-gray-200 "
>
  <div
    className="max-w-screen-xl flex flex-wrap items-center justify-between mx-auto p-4"
  >
    <span
      className="self-center text-2xl font-semibold whitespace-nowrap "
    >
      Chat App
    </span>
    <span>
      You are not logged in
    </span>
    <div
      className="hidden w-full md:block md:w-auto"
      id="navbar-dropdown"
    >
      <ul
        className="flex flex-col font-medium p-4 md:p-0 mt-4 border border-gray-100 rounded-lg bg-gray-50 md:flex-row md:space-x-8 md:mt-0 md:border-0 md:bg-white"
      >
        <li>
          <a
            className="block py-2 pl-3 pr-4 text-gray-900 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="block py-2 pl-3 pr-4 text-gray-900 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0"
            href="/inbox"
            onClick={[Function]}
          >
            Inbox
          </a>
        </li>
        <li>
          <a
            className="block py-2 pl-3 pr-4 text-gray-900 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
        <li>
          <a
            className="block py-2 pl-3 pr-4 text-gray-900 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
